Sources/MapboxDirections/Waypoint.swift

```diff
@@ -176,6 +180,15 @@ public class Waypoint: Codable, ForeignMemberContainerClass {
      */
     public var snappedDistance: LocationDistance?
     
+    /**
+     The layer of road that the waypoint is positioned which is used to filter the road segment that the waypoint will be placed on in Z-order. It is useful for avoiding ambiguity in the case of multi-level roads (such as a tunnel under a road).
+     
+     This property corresponds to the ['layers'](https://docs.mapbox.com/api/navigation/directions/#optional-parameters) query parameter in the Mapbox Directions API. If a matching layer is not found, the Mapbox Directions API will choose a suitable layer according to the other procided parameters.
```

Comment:
Although there’s no guarantee that the Directions API returns data from OpenStreetMap in every situation, the [OSM Wiki’s `layer` key documentation](https://wiki.openstreetmap.org/wiki/Key:layer) is probably the best explanation available for what goes into this property, so we should mention it somehow in the documentation comment. Even if the developer using the navigation SDK should just plumb the value from a PassiveLocationManager into a NavigationRouteOptions, it helps to know what these values correspond to in terms of real-world phenomena. There are some nuances that are difficult to discover otherwise, such as building passages being 0 rather than −1.